### ISSUE TITLE
crash while opening the file

### ISSUE THREAD

**Issue by iam-git** [HaikuArchives/BePDF#100]

**Comment by KevinAdams05** [comment]
Is anyone still having issues with this, or are there more details? It's not crashing for me on hrev59729 64bit.

<img width="949" height="774" alt="Image" src="https://github.com/user-attachments/assets/8e6834d7-c36b-445f-99a2-3b6ba51fb656" />

**Comment by korli** [comment]
> hrev59729

did you also test on r1beta5?

**Comment by KevinAdams05** [comment]
I was able to recreate this. It works fine on 64 bit, both Beta5 and the latest nightly. But it does crash on latest nightly 32bit. I will take a look.

**Comment by KevinAdams05** [comment]
The PDF parser is failing parsing the annotation links in this file. The /A (action) value is parsed as empty (but gets parsed correctly on the 64bit version). So we wind up with getting a null passed into LinkToString(). 

Adding a NULL resolves the issue, well at least it prevents the crash and allows the PDF to be opened:

  ```cpp
  void
  PDFView::LinkToString(LinkAction* action, BString* string) {
      if (action == NULL) {
          *string = "";
          return;
      }
      // rest of code is unchanged
  }
  ```


I am thinking this PDF has some non-standard encoding that xpdf on 32bit does not like. It processes other PDFs with links just fine on 32bit Haiku. Example of working PDF with links: [Intel SDM Vol2.pdf](https://github.com/user-attachments/files/28161443/Intel.SDM.Vol2.pdf)

I did not dig deep into the xpdf code yet, because it is quite old so I don't know if we want to fix bugs in it. I am happy to do so if that's the general philosophy.

**Comment by korli** [comment]
You might also log some warning, referencing this issue. This way this won't get completely ignored if a problem appears.

**Comment by KevinAdams05** [comment]
Makes sense. For now I will fix the crash and log an error. I'll create another issue for the underlying parser error and tag it with the xpdf tag. I'll look at that, and the other xpdf issues at a later point.

**Comment by korli** [comment]
Fixed by a807189f250080ba0b04f455d670cde3e0ba7c1b
